src/Compilers/Core/Portable/CommandLine/EditorConfig.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a single EditorConfig file, see http://editorconfig.org for details about the format.
+    /// </summary>
+    internal sealed class EditorConfig
+    {
+        // Matches EditorConfig section header such as "[*.{js,py}]", see http://editorconfig.org for details
+        private static readonly Regex s_sectionMatcher = new Regex(@"^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$", RegexOptions.Compiled);
+        // Matches EditorConfig property such as "indent_style = space", see http://editorconfig.org for details
+        private static readonly Regex s_propertyMatcher = new Regex(@"^\s*([\w\.\-_]+)\s*[=:]\s*(.*?)\s*([#;].*)?$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// A set of keys that are reserved for special interpretation for the editorconfig specification.
+        /// All values corresponding to reserved keys in a (key,value) property pair are always lowercased
+        /// during parsing.
+        /// </summary>
+        /// <remarks>
+        /// This list was retrieved from https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties
+        /// at 2018-04-21 19:37:05Z. New keys may be added to this list in newer versions, but old ones will
+        /// not be removed.
+        /// </remarks>
+        public static ImmutableHashSet<string> ReservedKeys { get; }
+            = ImmutableHashSet.CreateRange(CaseInsensitiveComparison.Comparer, new[] {
+                "root",
+                "indent_style",
+                "indent_size",
+                "tab_width",
+                "end_of_line",
+                "charset",
+                "trim_trailing_whitespace",
+                "insert_final_newline",
+            });
+
+        /// <summary>
+        /// A set of values that are reserved for special use for the editorconfig specification
+        /// and will always be lower-cased by the parser.
+        /// </summary>
+        public static ImmutableHashSet<string> ReservedValues { get; }
```

Comment:
>{ get; } [](start = 62, length = 8)

Nit: `=>`

Reply:
That would create a new set of ReservedValues on every call. This is just a readonly property.